cocotb/_sim_versions.py

```diff
@@ -0,0 +1,117 @@
+# Copyright cocotb contributors
+# Licensed under the Revised BSD License, see LICENSE for details.
+# SPDX-License-Identifier: BSD-3-Clause
+
+"""
+Classes to compare simulation versions.
+
+These are for cocotb-internal use only.
+Warning: these classes silently allow comparing versions of different simulators.
```

Comment:
I know it's not in sphinx, but:
```suggestion

.. warning::
    these classes silently allow comparing versions of different simulators.
```